src/hooks/useProfileModalControls.ts

```diff
@@ -0,0 +1,33 @@
+import { useCallback } from "react";
+
+import { User } from "types/User";
+
+import { WithId } from "utils/id";
+import { userProfileSelector } from "utils/selectors";
+
+import { setUserProfileData } from "store/actions/UserProfile";
+
+import { useSelector } from "./useSelector";
+import { useDispatch } from "./useDispatch";
+
+export const useProfileModalControls = () => {
+  const dispatch = useDispatch();
+  const selectedUserProfile = useSelector(userProfileSelector);
+
+  const openUserProfileModal = useCallback(
+    (userProfile?: WithId<User>) => {
+      dispatch(setUserProfileData(userProfile));
+    },
+    [dispatch]
+  );
+
+  const closeUserProfileModal = useCallback(() => {
+    dispatch(setUserProfileData(undefined));
+  }, [dispatch]);
+
+  return {
+    selectedUserProfile,
+    openUserProfileModal,
+    closeUserProfileModal,
+  };
```

Comment:
You gave https://github.com/sparkletown/sparkle/pull/1218#discussion_r592918721 a 👍🏻 and yet don't seem to have actually implemented it here at all? Usually a 👍🏻 lie that would imply also making the suggested change, and if you're not going to make it, then comments/further discussion as to why not.

```suggestion
  return {
    selectedUserProfile,
    hasSelectedUserProfile = selectedUserProfile !== undefined,
    openUserProfileModal,
    closeUserProfileModal,
  };
```